In [ ]:
library(tidyverse)
library(dplyr)

For this project, we were given the dataset "players.csv". This dataset has 196 observations and 7 variables. These variables include:

- experience: string, Categorical classification of the skill level of the player. Includes "Beginner", "Amateur", "Regular", "Pro", "Veteran"
- subscribe: logical, TRUE if subscribed, FALSE if not
- hashedEmail: string, unique email for each player
- played_hours: integer, total number of hours played
- name: string, name of player
- gender: string, gender of player
- Age: integer, age of player

For the data analysis, the name and hashed email of the player aren't helpful since we cannot use them to classify or predict anything about the players.

In [9]:
players <- read_csv("data/players.csv") |>
            select(experience, subscribe, played_hours, gender, Age)

averages <- players |> 
            select(played_hours, Age) |>
            map(mean, na.rm = TRUE)
averages

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


$played_hours
[1] 5.845918

$Age
[1] 21.13918

I have created a variable to hold the information from the dataset in players, The average amount of hours played is 5.85 hours. The average age of the players is 21 years old. Below I have the summary statistics of the gender of the players. A majority of the players are male, with 124 of the 196 players being male. This equates to 63% of the player base being male.

In [7]:
num_gender <- players |>
                count(gender) |>
                mutate(percentage = n / sum(n) * 100)
num_gender

gender            n   percentage
1 Agender             2  1.0204082
2 Female             37 18.8775510
3 Male              124 63.2653061
4 Non-binary         15  7.6530612
5 Other               1  0.5102041
6 Prefer not to say  11  5.6122449
7 Two-Spirited        6  3.0612245

The same analysis can be done for the experience level of the players. Amateur is the experience level with the most amount of players, with Pro being the least.

In [10]:
num_exp <- players |>
                count(experience) |>
                mutate(percentage = n / sum(n) * 100)
num_exp

experience n  percentage
1 Amateur    63 32.142857 
2 Beginner   35 17.857143 
3 Pro        14  7.142857 
4 Regular    36 18.367347 
5 Veteran    48 24.489796